File: backend/src/main/java/cn/etarch/mao/session/service/SessionCompactionEventService.java
package cn.etarch.mao.session.service;

import cn.etarch.mao.session.entity.SessionCompactionEvent;
import cn.etarch.mao.session.mapper.SessionCompactionEventMapper;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class SessionCompactionEventService {

    private final SessionCompactionEventMapper eventMapper;

    public SessionCompactionEvent record(Long sessionId,
                                         String triggerMode,
                                         long prevBoundaryMsgId,
                                         long boundaryMsgId,
                                         int compactedMessageCount,
                                         Integer promptTokens,
                                         Integer cachedTokens,
                                         Integer completionTokens,
                                         int summaryTokens,
                                         int savedTokens,
                                         long durationMs,
                                         String compactModel) {
        SessionCompactionEvent event = new SessionCompactionEvent();
        event.setSessionId(sessionId);
        event.setTriggerMode(triggerMode);
        event.setPrevBoundaryMsgId(prevBoundaryMsgId);
        event.setBoundaryMsgId(boundaryMsgId);
        event.setCompactedMessageCount(compactedMessageCount);
        event.setPromptTokens(promptTokens);
        event.setCachedTokens(cachedTokens);
        event.setCompletionTokens(completionTokens);
        event.setSummaryTokens(summaryTokens);
        event.setSavedTokens(savedTokens);
        event.setDurationMs(durationMs);
        event.setCompactModel(compactModel);
        eventMapper.insert(event);
        return event;
    }

    public List<SessionCompactionEvent> listBySessionId(Long sessionId) {
        return eventMapper.selectBySessionId(sessionId);
    }

    public int deleteBySessionId(Long sessionId) {
        return eventMapper.deleteBySessionId(sessionId);
    }
}

File: backend-ts/src/session/session-compaction-event.service.ts
import type { SessionCompactionEvent } from './types.js';
import type { SessionCompactionEventRepository } from './session-compaction.repository.js';

export class SessionCompactionEventService {
  constructor(private readonly repo: SessionCompactionEventRepository) {}

  async record(
    sessionId: number,
    triggerMode: string,
    prevBoundaryMsgId: number,
    boundaryMsgId: number,
    compactedMessageCount: number,
    promptTokens: number | null,
    cachedTokens: number | null,
    completionTokens: number | null,
    summaryTokens: number,
    savedTokens: number,
    durationMs: number,
    compactModel: string | null,
  ): Promise<SessionCompactionEvent> {
    const event: SessionCompactionEvent = {
      sessionId,
      triggerMode,
      prevBoundaryMsgId,
      boundaryMsgId,
      compactedMessageCount,
      promptTokens,
      cachedTokens,
      completionTokens,
      summaryTokens,
      savedTokens,
      durationMs,
      compactModel,
    };
    await this.repo.insert(event);
    return event;
  }

  listBySessionId(sessionId: number): Promise<SessionCompactionEvent[]> {
    return this.repo.selectBySessionId(sessionId);
  }

  deleteBySessionId(sessionId: number): Promise<number> {
    return this.repo.deleteBySessionId(sessionId);
  }
}
